src/js/utils/MesosStateUtil.js

```diff
@@ -2,8 +2,34 @@ import Util from './Util';
 
 const RESOURCE_KEYS = ['cpus', 'disk', 'mem'];
 
+function setSchedulerName(name, tasks) {
+  return tasks.map(function (task) {
+    let newTask = Object.assign({}, task);
+
+    newTask.scheduler = name;
+
+    return newTask;
+  });
+}
+
 const MesosStateUtil = {
 
+  assignSchedulerNameToTasks(state) {
```

Comment:
Hey @mlunoe, the thing is that there's an upcomming feature when we would like to i.e. draw a little icon next to the task identifying which scheduler has launched it. We actually started exactly from `isStartedByMarathon` flag approach. It was, again, made by name matching. I agree that idally we should get this info from the backend. If it's critical we can go back to `isStartedByMarathon` and matching by framework_id to solve that specific issue with checkboxes.